src/mds/Server.cc

```diff
@@ -5794,14 +5794,14 @@ int Server::parse_quota_vxattr(string name, string value, quota_info_t *quota)
       }
     } else if (name == "quota.max_bytes") {
       /*
-       * The "quota.max_bytes" must be aligned to 4MB if greater than or
-       * equal to 4MB, otherwise must be aligned to 4KB.
+       * The "quota.max_bytes" must be aligned to 4MiB if greater than or
+       * equal to 4MiB, otherwise must be aligned to 4KiB.
        */
       string cast_err;
       int64_t q = strict_iec_cast<int64_t>(value, &cast_err);
       if(!cast_err.empty() ||
-         (!IS_ALIGNED(q, CEPH_4M_BLOCK_SIZE) &&
-          (q < CEPH_4M_BLOCK_SIZE && !IS_ALIGNED(q, CEPH_4K_BLOCK_SIZE)))) {
+         (q >= CEPH_4M_BLOCK_SIZE && !IS_ALIGNED(q, CEPH_4M_BLOCK_SIZE)) ||
+         (q < CEPH_4M_BLOCK_SIZE && !IS_ALIGNED(q, CEPH_4K_BLOCK_SIZE))) {
         dout(10) << __func__ << ":  failed to parse quota.max_bytes: "
                  << cast_err << dendl;
         return -CEPHFS_EINVAL;
```

Comment:
this looks a bit incorrect to me. If in case `cast_err` is empty and the checks for block size encounter invalid alignment, it would log like this
```
2023-04-05T19:46:49.073+0000 7f2e377a6700 10 mds.0.server parse_quota_vxattr:  failed to parse quota.max_bytes: 
```
i.e. log without no reason why it failed.

Instead what can be done is to split this up in two if statements, something like

```
      if(!cast_err.empty()) {
        dout(10) << ":  failed to parse quota.max_bytes: "
                 << cast_err << dendl;
        return -CEPHFS_EINVAL;
      }
      if(!IS_ALIGNED(q, CEPH_4K_BLOCK_SIZE) || !IS_ALIGNED(q, CEPH_4M_BLOCK_SIZE)) {
        if (q < CEPH_4K_BLOCK_SIZE) {
          dout(10) << __func__ << ": failed to parse quota.max_bytes," 
                   << " size must be aligned to 4KiB"<< dendl;
        } else {
          dout(10) << __func__ << ": failed to parse quota.max_bytes," 
                   << " size must be aligned to 4MiB"<< dendl;
        }
        return -CEPHFS_EINVAL;
      }
      
```

Reply:
Looks good to me and will fix it.